membership-attribute-service/app/services/stripe/StripeService.scala

```diff
@@ -6,9 +6,17 @@ import com.gu.stripe.StripeServiceConfig
 import com.gu.zuora.api.{PaymentGateway, RegionalStripeGateways}
 
 import scala.concurrent.{ExecutionContext, Future}
+import com.gu.zuora.api.StripeTortoiseMediaPaymentIntentsMembershipGateway
 
-class StripeService(apiConfig: StripeServiceConfig, val basicStripeService: BasicStripeService) {
-  val paymentIntentsGateway: PaymentGateway = RegionalStripeGateways.getPaymentIntentsGatewayForCountry(apiConfig.stripeAccountCountry)
+class StripeService(
+    apiConfig: StripeServiceConfig,
+    val basicStripeService: BasicStripeService,
+) {
+  val paymentIntentsGateway: PaymentGateway =
+    apiConfig.variant match {
+      case Some("tortoise-media") => StripeTortoiseMediaPaymentIntentsMembershipGateway
+      case _ => RegionalStripeGateways.getPaymentIntentsGatewayForCountry(apiConfig.stripeAccountCountry)
+    }
```

Comment:
does seem a bit odd that we are first doing it off the "variant" (which includes country) and if that fails we use the country itself.
I think for consistency we should just change it to use variant for everything and get rid of the `getPaymentIntentsGatewayForCountry` function, then it's all clear and in one place:
```suggestion
    apiConfig.variant match {
      case "tortoise-media" => StripeTortoiseMediaPaymentIntentsMembershipGateway
      case "au-membership" => StripeAUPaymentIntentsMembershipGateway
      case _ => StripeUKPaymentIntentsMembershipGateway
    }
```
Since we're pattern matching it, it might also make sense to turn variant [into an ADT ](https://www.baeldung.com/scala/algebraic-data-types#sum-types) but I don't think there's a strong argument yet as it's mostly for config.

Reply:
true but if you look in that function then it's just an if statement so we're just flattening it up a level.
Also I know that for recurring we have one AU account, one ROW account, and one tortioise account so it makes sense to represent that in the code.